lib/chef/mixin/powershell_exec.rb

```diff
@@ -104,22 +104,24 @@ module PowershellExec
       #
       # @param script [String] script to run
       # @param interpreter [Symbol] the interpreter type, `:powershell` or `:pwsh`
+      # @param timeout [Integer, nil] timeout in seconds.
       # @return [Chef::PowerShell] output
-      def powershell_exec(script, interpreter = :powershell)
-        case interpreter
+      def powershell_exec(script, options = {})
+        timeout = options.fetch(:timeout, nil)
+        case options.fetch(:interpreter, :powershell)
         when :powershell
-          Chef::PowerShell.new(script)
+          Chef::PowerShell.new(script, timeout: timeout)
         when :pwsh
-          Chef::Pwsh.new(script)
+          Chef::Pwsh.new(script, timeout: timeout)
         else
           raise ArgumentError, "Expected interpreter of :powershell or :pwsh"
         end
       end
 
       # The same as the #powershell_exec method except this will raise
       # Chef::PowerShell::CommandFailed if the command fails
-      def powershell_exec!(script, interpreter = :powershell)
```

Comment:
don't we still want the ps interpreter?

Reply:
yep it is available. the options here get pass directly to `powershell_exec` call instead of individually writing down all parameters and then again passing them to the method call. https://github.com/chef/chef/blob/b2d46ab518ae79c892ff14c6e78a09acae03883a/lib/chef/mixin/powershell_exec.rb#L123-L124
this looked cleaner than doing
```ruby
def powershell_exec!(script, interpreter: :powershell, timeout: -1)
         cmd = powershell_exec(script,  interpreter: interpreter, timeout: timeout)
```
we only read the values in `powershell_exec` so passing It as a hash argument made it a bit cleaner https://github.com/chef/chef/blob/b2d46ab518ae79c892ff14c6e78a09acae03883a/lib/chef/mixin/powershell_exec.rb#L109-L112

Reply:
I think the switch from `interpreter = :powershell` to `interpreter: :powershell` is likely a breaking change.

I'm going to see if I can revert that change and ship it quickly since this is now blocking john's work.

Reply:
Also the `options = {}` shouldn't be used any more in favor of proper kwargs since it behaves differently on ruby-3.0 I think.  It is also a lot easier to see the API on the method signature rather than having to look through the code (and messy method signatures are good feedback on the API being messy -- although a lot of times there's just nothing we can do about that).